src/hooks/users.ts

```diff
@@ -1,63 +1,88 @@
-import { useMemo } from "react";
-import { isLoaded } from "react-redux-firebase";
-
-import { User } from "types/User";
-import { ValidStoreAsKeys } from "types/Firestore";
-
-import { usersSelector, usersByIdSelector } from "utils/selectors";
+import { worldUsersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
-import { useSelector } from "hooks/useSelector";
-import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
-import { useVenueId } from "hooks/useVenueId";
-import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "./useSelector";
+import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+import { useVenueId } from "./useVenueId";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
 
-const useConnectVenueUsers = () => {
+import { User } from "types/User";
+import { useMemo } from "react";
+
+export const useConnectWorldUsers = () => {
   const venueId = useVenueId();
 
-  // @debt refactor this + related code so as not to rely on using a shadowed 'storeAs' key
-  //   this should be something like `storeAs: "currentVenueUsers"` or similar
-  useFirestoreConnect(
-    venueId
-      ? {
-          collection: "users",
-          where: ["enteredVenueIds", "array-contains", venueId],
-          storeAs: "users" as ValidStoreAsKeys, // @debt super hacky, but we're consciously subverting our helper protections
-        }
-      : undefined
+  const { isCurrentVenueLoaded, currentVenue } = useConnectCurrentVenueNG(
+    venueId!
   );
-};
 
-export const useVenueUsers = (): readonly WithId<User>[] => {
-  useConnectVenueUsers();
+  useFirestoreConnect(() => {
+    if (!isCurrentVenueLoaded || !currentVenue) return [];
+
+    const relatedLocationIds = [currentVenue.id];
 
-  return useSelector(usersSelector) ?? [];
+    if (currentVenue.parentId) {
+      relatedLocationIds.push(currentVenue.parentId);
+    }
+
+    return [
+      {
+        collection: "users",
+        where: ["enteredVenueIds", "array-contains-any", relatedLocationIds],
+        storeAs: "worldUsers",
+      },
+    ];
+  });
 };
 
-export const usePartygoers = (): readonly WithId<User>[] => {
-  const lastSeenThreshold = useUserLastSeenThreshold();
+export const useWorldUsers = (): {
```

Comment:
As called out in a few places (but probably not all of them), the implementation of `useWorldUsers` here doesn't satisfy that same logic that `usePartygoers` used to, as `usePartygoers` filtered down to only the venue users via `useVenueUsers`, which we no longer have here.

I suggest implementing `useRecentVenueUsers` as described elsewhere in my comments, then going back and changing all of the places that `usePartygoers` used to be to use that, rather than those places using `useRecentWorldUsers`